receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -157,9 +157,8 @@ func createKafkaClient(config Config) (sarama.ConsumerGroup, error) {
 	return sarama.NewConsumerGroup(config.Brokers, config.GroupID, saramaConfig)
 }
 
-func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) error {
-	ctx, cancel := context.WithCancel(context.Background())
-	c.cancelConsumeLoop = cancel
+func (c *kafkaTracesConsumer) Start(ctx context.Context, host component.Host) error {
+	ctx, c.cancelConsumeLoop = context.WithCancel(ctx)
```

Comment:
```suggestion
func (c *kafkaTracesConsumer) Start(ctx context.Context, host component.Host) error {
	ctx, c.cancelConsumeLoop = context.WithCancel(ctx)
```
I don't think this is the solution here, as it goes against context usage for components. The comment about `Start` [here](https://pkg.go.dev/go.opentelemetry.io/collector/component#Component) says we shouldn't use the passed in context for long-running background operations. At the moment using the passed in context doesn't cause any issues, but we don't want to go against the spec unless there's good reason.

Also, I don't believe this will resolve any kind of stuck shutdown. `context.WithCancel` creates a child context that can be cancelled independently of the parent. If the parent is shutdown, the child will automatically be shutdown as well, but there's no requirement vice versa. (The child can be shutdown just fine even if the parent is still running.)

I think there's else going on here, but feel free to correct me if I'm missing something here, or if you tested and confirmed this resolved the issue 🙂